src/parser/to_parse.rs

```diff
@@ -0,0 +1,391 @@
+use std::io::Read;
+use std::mem::MaybeUninit;
+
+use byteorder::{BigEndian, ReadBytesExt};
+
+use super::Error;
+
+pub const ALIGNMENT: usize = 4;
+
+#[allow(unused)]
+pub trait ToParse: Sized {
+    fn parse<R: Read>(src: &mut R) -> Result<Self, Error>;
+}
+
+#[allow(unused)]
+pub fn parse<T: ToParse, R: Read>(src: &mut R) -> Result<T, Error> {
+    T::parse(src)
+}
+
+pub fn read_padding<R: Read>(src: &mut R, n: usize) -> Result<(), Error> {
+    let mut buf = [0u8; ALIGNMENT];
+    let padding = (ALIGNMENT - n % ALIGNMENT) % ALIGNMENT;
+    src.read_exact(&mut buf[..padding]).map_err(|_| Error::IncorrectPadding)
+}
+
+impl ToParse for u32 {
+    fn parse<R: Read>(src: &mut R) -> Result<Self, Error> {
```

Comment:
why not just

```
fn u32(srd: &mut R) -> Result<Self, Error> {
     src.read_u32::<BigEndian>().map_err(|_| Error::IO)
}
```

for what trait?

Here we can return exactly std::io::Result actually